internal/controller/nopresource/nopresource.go

```diff
@@ -128,13 +97,70 @@ type external struct {
 }
 
 func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
-	cr, ok := mg.(*v1alpha1.MyType)
+	cr, ok := mg.(*v1alpha1.NopResource)
 	if !ok {
-		return managed.ExternalObservation{}, errors.New(errNotMyType)
+		return managed.ExternalObservation{}, errors.New(errNotNopResource)
+	}
+	startTime := cr.CreationTimestamp
+
+	// If object was deleted, return it does not exist so that managed reconciler removes finalizer
+	if meta.WasDeleted(mg) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+
+	// Logic 1 - Set condition as soon as the time elapsed in seconds is equal to time specified in Spec
+	// for i := 0; i < len(cr.Spec.ForProvider.ConditionAfter); i++ {
+	// 	elapsed, _ := time.ParseDuration(cr.Spec.ForProvider.ConditionAfter[i].Time)
+	// 	if time.Duration(time.Since(startTime.Time).Seconds()) == time.Duration(elapsed.Seconds()) {
+	// 		fmt.Printf("Calling update on index %d\n", i)
+
+	// 		x := runtimev1alpha1.Condition{
+	// 			Type:               runtimev1alpha1.ConditionType(cr.Spec.ForProvider.ConditionAfter[i].ConditionType),
+	// 			Status:             v1.ConditionStatus(cr.Spec.ForProvider.ConditionAfter[i].ConditionStatus),
+	// 			LastTransitionTime: metav1.Now(),
+	// 			Reason:             runtimev1alpha1.ReasonAvailable,
+	// 		}
+
+	// 		cr.Status.SetConditions(x)
+
+	// 	}
+	// }
+
+	// Logic 2 - Set condition at every reconcile to the latest specified condition in Spec
```

Comment:
This logic will set conditions at every reconcile to the latest specified condition in Spec. 
The behavior of the same can be found at logs/logic2. log